File: frontend/src/types/knowledge-base.ts
export interface KnowledgeBaseItem {
  id: string;
  name: string;
  description: string;
  document_count: number;
  created_at: string;
  updated_at: string;
}

export interface KnowledgeDocumentItem {
  id: string;
  knowledge_base_id: string;
  file_name: string;
  file_size: number;
  page_count: number;
  chunk_count: number;
  parse_status: "processing" | "ready" | "failed" | string;
  parse_error: string;
  created_at: string;
  updated_at: string;
}

export interface KnowledgeChatTurn {
  role: "user" | "assistant";
  content: string;
}

export interface KnowledgeContextChunk {
  chunk_id: string;
  document_id: string;
  document_name: string;
  chunk_index: number;
  content: string;
  score?: number | null;
}

export interface KnowledgeChatResponse {
  answer: string;
  context_chunks: KnowledgeContextChunk[];
}

File: backend/app/schemas/knowledge_base.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class KnowledgeBaseCreateRequest(BaseModel):
    name: str = Field(min_length=1, max_length=128)
    description: str = Field(default="", max_length=2000)


class KnowledgeBaseOut(BaseModel):
    id: str
    name: str
    description: str
    document_count: int
    created_at: datetime
    updated_at: datetime


class KnowledgeDocumentOut(BaseModel):
    id: str
    knowledge_base_id: str
    file_name: str
    file_size: int
    page_count: int
    chunk_count: int
    parse_status: str
    parse_error: str
    created_at: datetime
    updated_at: datetime


class KnowledgeChatTurn(BaseModel):
    role: str = Field(pattern="^(user|assistant)$")
    content: str = Field(min_length=1)


class KnowledgeBaseChatRequest(BaseModel):
    message: str = Field(min_length=1)
    history: list[KnowledgeChatTurn] = Field(default_factory=list)
    top_k: int = Field(default=4, ge=1, le=10)
    llm_provider: str | None = None
    llm_model: str | None = None


class KnowledgeContextChunk(BaseModel):
    chunk_id: str
    document_id: str
    document_name: str
    chunk_index: int
    content: str
    score: float | None = None


class KnowledgeBaseChatResponse(BaseModel):
    answer: str
    context_chunks: list[KnowledgeContextChunk]
